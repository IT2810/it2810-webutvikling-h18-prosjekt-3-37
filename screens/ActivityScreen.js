import React from 'react';
import { StyleSheet, View } from 'react-native';
import { createStackNavigator } from 'react-navigation';

import { Container, Header, Title, Content, Footer, FooterTab, Button, Left, Right, Body, Icon, Text, List, ListItem, CheckBox } from 'native-base';



class ActivityScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      completedTasks: 7,
      totalTasks: 12,
      todos: [{name: null, checked: null}, {name: "Label 2", checked: true}, {name: "Label 3", checked: false}, {name: "Label 4", checked: true}, {name: "Label 5 ", checked: true}]
    };
  }

  setTodos(newTodo) {
    let tempTodos = this.state.todos.slice();
    tempTodos.push(newTodo);
    this.setState({
      todos: tempTodos
    });
  }

  updateTaskCount() {
    let completedCount = 0;
    let totaltCount = this.state.todos.length;
    for (let i = 0; i < this.state.todos.length; i++) {
      if (this.state.todos[i].checked) {
        completedCount++;
      }
    }
    this.setState({
      completedTasks: completedCount,
      totalTasks: totaltCount
    })
  }

  handleClickCheckbox (index) {
    let tempTodos = this.state.todos.slice();
    tempTodos[index].checked = !tempTodos[index].checked;
    this.setState({
      todos: tempTodos
    })
    this.updateTaskCount();
  }

  render() {
    return (
      <Container>
        <Content contentContainerStyle={styles.content}>

        </Content>
        <Footer >
          <FooterTab style={styles.footer}>
            <Button full>
              <Text>Gruppe 37</Text>
            </Button>
          </FooterTab>
        </Footer>
      </Container>
    );
  }
}

export default createStackNavigator({
    Activities: {
      screen: ActivityScreen,
      navigationOptions: ({ navigation }) => ({
        title: "Activities",
        headerLeft: <Icon style={styles.menuIcon} name="menu" size={50} onPress={() => navigation.toggleDrawer() } />,
        headerStyle: {
          backgroundColor: "#645CFF",
        },
        headerTitleStyle: {
          color: "white"
        }
      })
    }
  },
  {
    initialRouteName: 'Activities',
  }
);

const styles = StyleSheet.create({
  content: {
    flex: 1,
    justifyContent: 'space-between',
  },
  menuIcon: {
    padding: 15,
    color: "white"
  },
footer: {
  backgroundColor: "#645CFF",
}
})
