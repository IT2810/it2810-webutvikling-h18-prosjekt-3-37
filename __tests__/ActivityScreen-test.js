// __tests__/Login-page-test.js
import 'react-native';
import React from 'react';
import ActivityScreen from '../screens/ActivityScreen';

import renderer from 'react-test-renderer';

it('works', () => {
  expect(true).toBeTruthy();
});

it('renders correctly', () => {
  jest.useFakeTimers();
  const tree = renderer.create(
    <ActivityScreen />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});