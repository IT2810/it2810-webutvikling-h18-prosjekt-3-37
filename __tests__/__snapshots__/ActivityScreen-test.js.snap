// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "paddingTop": 64,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
              "height": 1334,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#F8F8F8",
                "borderColor": "#cbcbcb",
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "center",
                "left": 0,
                "paddingBottom": 0,
                "right": 0,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": undefined,
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                  Object {
                    "backgroundColor": "#645CFF",
                  },
                ]
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": null,
                    "borderColor": null,
                    "borderLeftWidth": null,
                    "borderRadius": 0,
                    "borderRightWidth": null,
                    "borderTopWidth": null,
                    "borderWidth": undefined,
                    "elevation": 0,
                    "flex": 1,
                    "flexDirection": null,
                    "height": 55,
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingBottom": 6,
                    "paddingTop": 6,
                    "shadowColor": null,
                    "shadowOffset": null,
                    "shadowOpacity": null,
                    "shadowRadius": null,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#6b6b6b",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "lineHeight": 16,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                    }
                  }
                  uppercase={false}
                >
                  Gruppe 37
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#645CFF",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "#645CFF",
              "borderBottomColor": "#A7A7AA",
              "borderBottomWidth": 0.5,
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "flexDirection": "row",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "left": 70,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 70,
                    "top": 0,
                  }
                }
              >
                <Text
                  accessibilityTraits="header"
                  accessible={false}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 17,
                      "fontWeight": "600",
                      "marginHorizontal": 16,
                      "textAlign": "center",
                    }
                  }
                >
                  Activities
                </Text>
              </View>
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "flexDirection": "row",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "top": 0,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
